### ISSUE TITLE
Some questions about the loss function of  iwacgan and the discriminator settings

### ISSUE THREAD

**Issue by weihancug** [shaform/DeepNetworks#1]
Hi your project is very detailed and easy to read. But I am a pytorch user, and can't understand it particularly clear.   Therefore，there some question I try to figure out.

If i have not made a mistake, you use the three discriminators in iwacgan to deal with real, fake, the interpolation samples respectively. I have read some iwgan code where  only one  discriminator is used. What is the difference between them? Will the three discriminators are updated at the same time according the loss function?@shaform

In addition, the generator loss and discriminator loss is 
'''python
self.g_total_loss = self.g_loss + self.g_c_loss + self.g_reg_loss

self.d_total_loss = (self.d_loss + self.d_c_loss + self.d_grad_loss+ self.d_reg_loss)
'''
The total loss function looks equal to a value that iwgan loss plus the category crossover entropy loss. Do not need any modification operation?

**Comment by weihancug** [comment]
I'm sorry that there are some problems with the inserted code

**Comment by shaform** [comment]
(1) The three discriminators share the same weights so they are effectively one discriminator. We could also put real, fake, and interpolation into a batch for a single discriminator, but I find it easier this way.
(2) Yes. Although one might think this is problematic because the scale of `d_loss` / `g_loss` is much greater then `c_loss`, surprisingly it works for MNIST.

**Comment by weihancug** [comment]
Thanks for your responses, I plan to use it for remote sensing images and observe the results
